src/Constraints/CollectionValidator.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Tuf\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\Constraints\Collection;
+use Symfony\Component\Validator\Constraints\CollectionValidator as SymfonyCollectionValidator;
+
+class CollectionValidator extends SymfonyCollectionValidator
+{
+
+    /**
+     * {@inheritdoc}
+     */
+    public function validate($value, Constraint $constraint)
+    {
+        if (!empty($constraint->unsupportedFields)) {
+            foreach ($constraint->unsupportedFields as $unsupportedField) {
+                $existsInArray = \is_array($value) && \array_key_exists($unsupportedField, $value);
+                $existsInArrayAccess = $value instanceof \ArrayAccess && $value->offsetExists($unsupportedField);
+                if ($existsInArray || $existsInArrayAccess) {
+                    $this->context->buildViolation('This field is not supported.')
+                      ->atPath('['.$unsupportedField.']')
```

Comment:
Is this supposed to be using single quotes? It looks like it should be double-quoted so that $unsupportedField will be replaced...

Reply:
Yep. I had just copied this from the [symfony lib](https://github.com/symfony/validator/blob/5.x/Constraints/CollectionValidator.php#L61). Fixing